### ISSUE TITLE
Missing register in landing page of registry

### ISSUE THREAD

**Issue by arantzaetxebarria** [ec-jrc/re3gistry#147]
From: [MartinTuchyna](https://github.com/MartinTuchyna)
[original issue](https://github.com/INSPIRE-MIF/helpdesk-registry/issues/29)

> Dear Re3gistry Team,
> 
> any idea, why we are not able to see this testing Register https://er.hany.sk/extensions
> also on the landing page of the Re3gistry? https://er.hany.sk/registry
> 
> Thanks in advance,
> 
> best regards, Martin

**Comment by arantzaetxebarria** [comment]
From: [emanuelaepure10](https://github.com/emanuelaepure10)

> Hi @MartinTuchyna & @MichalMed ,
> 
> Can I ask you how you have create the register?
> From the management interface?
> 
> Do you see it in the management interface as a part of the registry?
> 
> if so, probably you need to remove the cache folder from you file system and try to reload the page.
> 
> Soon we have released a new version of the software which is updating the cache once 'something' has been change on the management interface.
> 
> Please let us know if removing the cache or after a update the problem has need solved
> 
> Thank you in advanced

**Comment by MartinTuchyna** [comment]
Dear Oscar,

seems we are facing the same issue also on [Docker instance of the registry](https://github.com/ec-jrc/re3gistry/issues/22#issuecomment-1280097726), where for instance, we have tried to create following two registers:

![obrázok](https://user-images.githubusercontent.com/6195936/196065000-e21fda91-aea9-4b17-9c87-76e6ab538735.png)

Everything looked ok, but there is nothing available [via landing page on the frontend](https://registry.stage.geocloud.sk/)

![obrázok](https://user-images.githubusercontent.com/6195936/196065063-bd3e389e-baad-491c-b195-adb60ac9a460.png)

Thanks, Martin

**Comment by MartinTuchyna** [comment]
PS: What is also interesting, that in public frontend landig page I can not change current default language version from Slovak to English:

https://registry.stage.geocloud.sk/

![obrázok](https://user-images.githubusercontent.com/6195936/196066483-c26ecf0e-aab4-4487-9f4d-2b159a5066d4.png)

![obrázok](https://user-images.githubusercontent.com/6195936/196066504-d32ddd64-3b80-4561-9d93-0440a26a5f4f.png)

When trying incognito / unlogeed access, situation is oposit. I can not change English to Slovak:

![obrázok](https://user-images.githubusercontent.com/6195936/196066593-fb3dfeb2-19b9-454f-84a0-5e988d0347d2.png)

![obrázok](https://user-images.githubusercontent.com/6195936/196066607-0667e3e1-b0e3-4dd5-ab0c-8d6eb7e26369.png)

![obrázok](https://user-images.githubusercontent.com/6195936/196066632-39523809-7805-4376-997e-e1d8652eeb01.png)

**Comment by oscar9** [comment]
Hi @MartinTuchyna ,

At first glance, these seem like two non-Docker related bugs, would you mind opening two new issues with them?
Also, have you tried the new version 2.4.0-rc1? There were changes in the language section and in the cookies that may affect these errors that you are commenting on.

Thanks in advance.

**Comment by MartinTuchyna** [comment]
Hi @oscar9 ,

yes, we have updated the version to the 2.4.0-rc1 ([@PetoP](https://github.com/PetoP)), although on the landing page still indicates 2.3.0
![obrázok](https://user-images.githubusercontent.com/6195936/200386741-c0d4a85b-221b-4a1f-aa62-31b0d8a9dc80.png)

and have to admit that this version comes with even more bugs, like for example, when trying to create first register initial page is splitted to two windows, whilst second does not allows any further actrion - see snap. 
![obrázok](https://user-images.githubusercontent.com/6195936/200386113-b00d0576-1b38-4f19-bead-ce965b76774e.png)

SK page for creating first register is missing completely:
![obrázok](https://user-images.githubusercontent.com/6195936/200386247-cdbb20d9-9583-4b88-86ab-9ad5dc0a35aa.png)

So instead the progress, we are back on beginning.

**Comment by oscar9** [comment]
Hi @MartinTuchyna 
I'm going to label this ticket as bug. We will double-check with the development team and see what is happening. We will inform you of any discovery.
Thanks

**Comment by jescriu** [comment]
Dear @MartinTuchyna,
Really sorry to hear about these issues. JRC will push for the prioritisation of their resolution by the contractor, who recently delivered both, Re3gistry 2.4.0-rc1 and 2.4.0.
By the way, I assume you are still using the ISO 19135 workflow for managing your registry, and not the recent simplified approach which was added in the last Re3gistry version - Could you please confirm?

**Comment by oscar9** [comment]
Hi @MartinTuchyna ,

1. The frontend works as a stand-alone application and the version of the registry that appears is manually modified. We are going to update that version in the repository because it was not modified during the last release, thanks for reporting that. This modification only involves modifying the value by hand in the different htmls of the webapp application . I take this opportunity to explain that by this I mean that this version is not obtained by calling the API with the re3gistry that it is using, the same frontend could run on different versions of the Registry at different times.
https://github.com/ec-jrc/re3gistry/tree/master/sources/Re3gistry2ServiceWebapp/public_html

2. It looks like there is a problem in the Apache httpd configuration. I've accessed the address in your image (https://registry.stage.geocloud.sk/) and looked at the calls it makes. I don't see it making any API calls or similar.

![image](https://user-images.githubusercontent.com/4819782/200777425-ff8dd18b-4ccf-4f97-9f06-72513a31a984.png)

For example, this is from a working example:
![image](https://user-images.githubusercontent.com/4819782/200777138-bf4c5b86-299e-47c1-be45-e8184995ac64.png)

Also, a possible test for this is, you could create a register or item and try to access that specific item (example similar to. https://registry.stage.geocloud.sk/myregister/item01 ) and also check if is working or if is just the main page that is failing.

3.We have created from scratch a new instance with master language EN, and languages EN,SK and vice-versa, master SK, languages EN,SK, but in any case we can replicate that error. Could you send us the tomcat logs when this error is ocurring?

We are testing all this over the final version 2.4.0, there were not major changes between this version and the rc1. 

Let us know any feedback from this

Best regards,
Óscar

**Comment by MartinTuchyna** [comment]
Dear @jescriu 

> By the way, I assume you are still using the ISO 19135 workflow for managing your registry, and not the recent simplified approach which was added in the last Re3gistry version - Could you please confirm?

Not sure, what is the difference between 2 approaches. Normaly, first step is to create the register under the Content tab.
Example of the new register for INSPIRE Mappings:

![obrázok](https://user-images.githubusercontent.com/6195936/201232585-a24ff761-781b-4b61-8343-3302c52ca911.png)

After filling the content and summary and pressing "Add register" next page is offered:

![obrázok](https://user-images.githubusercontent.com/6195936/201231973-c11f73e2-e9b6-4971-954a-abadfb1aa3a7.png)

but there is none option to continue, as you can see on the snap.

If I remember, above elements used to be on the first page, therefore this behaviour is quite strange.

@oscar9 

As you can see, I am not able to create any item now. Anyway I will try to find out, if I can get the tomcat log.

Best regards, Martin

**Comment by PetoP** [comment]
Hi all,

I have updated all system components based on current state in this repo. I have also modified our docker-compose file to reflect all changes made in repository (our only change now are changed ports as we already have few of them occupied).

2. There was problem with permissions on the server. After fixing it more resources are loaded  (now only logos pngs and one css files are missing). Overall state seem unchanged however.

3.  @oscar9, following is tomcat log as you requested. No user actions are captured in it, if you are missing some, please let me know what should I test to do in order to get it into log file.

```
Listening for transport dt_socket at address: 8000
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Server version name:   Apache Tomcat/7.0.109
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Server built:          Apr 22 2021 18:43:38 UTC
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Server version number: 7.0.109.0
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: OS Name:               Linux
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: OS Version:            3.10.0-1127.19.1.el7.x86_64
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Architecture:          amd64
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Java Home:             /opt/java/openjdk/jre
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: JVM Version:           1.8.0_292-b10
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: JVM Vendor:            AdoptOpenJDK
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: CATALINA_BASE:         /usr/local/tomcat
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: CATALINA_HOME:         /usr/local/tomcat
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -Djava.util.logging.config.file=/usr/local/tomcat/conf/logging.properties
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -agentlib:jdwp=transport=dt_socket,address=8000,server=y,suspend=n
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -Dignore.endorsed.dirs=
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -Dcatalina.base=/usr/local/tomcat
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -Dcatalina.home=/usr/local/tomcat
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.VersionLoggerListener log
INFO: Command line argument: -Djava.io.tmpdir=/usr/local/tomcat/temp
Nov 11, 2022 5:50:40 PM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
INFO: Loaded Apache Tomcat Native library [1.2.28] using APR version [1.6.5].
Nov 11, 2022 5:50:40 PM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
INFO: APR capabilities: IPv6 [true], sendfile [true], accept filters [false], random [true].
Nov 11, 2022 5:50:40 PM org.apache.catalina.core.AprLifecycleListener initializeSSL
INFO: OpenSSL successfully initialized [OpenSSL 1.1.1f  31 Mar 2020]
Nov 11, 2022 5:50:40 PM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-apr-8080"]
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.Catalina load
INFO: Initialization processed in 816 ms
Nov 11, 2022 5:50:40 PM org.apache.catalina.core.StandardService startInternal
INFO: Starting service [Catalina]
Nov 11, 2022 5:50:40 PM org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet Engine: Apache Tomcat/7.0.109
Nov 11, 2022 5:50:40 PM org.apache.catalina.startup.HostConfig deployDirectory
INFO: Deploying web application directory [/usr/local/tomcat/webapps/re3gistry2]
Nov 11, 2022 5:50:43 PM org.apache.catalina.startup.TldConfig execute
INFO: At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
[localhost-startStop-1] INFO org.apache.shiro.web.env.EnvironmentLoader - Starting Shiro environment initialization.
[localhost-startStop-1] INFO org.apache.shiro.config.IniSecurityManagerFactory - Realms have been explicitly set on the SecurityManager instance - auto-setting of realms will not occur.
[localhost-startStop-1] INFO org.apache.shiro.web.env.EnvironmentLoader - Shiro environment initialized in 322 ms.
### Initializing the system ...
File name properties path: configurations_files/configuration.properties
78 properties loaded from configurations_files/configuration.properties
### Setting up the logger
### The system is now initialized. Path for the helper files: /usr/local/tomcat/webapps/re3gistry2/WEB-INF
Nov 11, 2022 5:50:44 PM org.apache.catalina.startup.HostConfig deployDirectory
INFO: Deployment of web application directory [/usr/local/tomcat/webapps/re3gistry2] has finished in [3,731] ms
Nov 11, 2022 5:50:44 PM org.apache.catalina.startup.HostConfig deployDirectory
INFO: Deploying web application directory [/usr/local/tomcat/webapps/re3gistry2restapi]
Nov 11, 2022 5:50:45 PM org.apache.catalina.startup.TldConfig execute
INFO: At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
### Initializing the system ...
File name properties path: configurations_files/configuration.properties
6 properties loaded from configurations_files/configuration.properties
### The system is now initialized. Path for the helper files: /usr/local/tomcat/webapps/re3gistry2restapi/WEB-INF/classes/configurations_files
[EL Info]: 2022-11-11 17:50:47.49--ServerSession(2005687168)--EclipseLink, version: Eclipse Persistence Services - 2.7.5.v20191016-ea124dd158
[EL Info]: connection: 2022-11-11 17:50:48.059--ServerSession(2005687168)--/file:/usr/local/tomcat/webapps/re3gistry2restapi/WEB-INF/classes/_Re3gistry2 login successful
Nov 11, 2022 5:50:48 PM org.apache.catalina.startup.HostConfig deployDirectory
INFO: Deployment of web application directory [/usr/local/tomcat/webapps/re3gistry2restapi] has finished in [3,833] ms
Nov 11, 2022 5:50:48 PM org.apache.coyote.AbstractProtocol start
INFO: Starting ProtocolHandler ["http-apr-8080"]
Nov 11, 2022 5:50:48 PM org.apache.catalina.startup.Catalina start
INFO: Server startup in 7699 ms
```

Best regards,

Peter

**Comment by oruscalleda** [comment]
Hi all,

A solution to this issue has been included in the pre release version published today. Please update your instance of the software an let us know if the error still happens.

Best regards,

Oriol

**Comment by MartinTuchyna** [comment]
Dear Oriol,

we have updated to the 2.4.1  version (although in HTML source code is still written 2.3.0).

After login to the backend, when trying to create the first register - these are the only options available:

![obrázok](https://user-images.githubusercontent.com/6195936/220104717-fcadd4cf-0e1b-4529-84f4-3fe00fcf9a00.png)

So we are not able to even start to create Register.

On the default frontend, we are not able to see even Example codelist preprepared in backend:
https://registry.stage.geocloud.sk/examplecodelist

![obrázok](https://user-images.githubusercontent.com/6195936/220105364-11504717-67e1-4ba3-9331-3e2748af2442.png)

**Comment by oruscalleda** [comment]
Hi @MartinTuchyna,

Sorry to hear that, I'll get into it and let you know when I discover something.

I can see from your print screen you seem to have something wrong on your configuration, as you can see the address for your registry has "http://host.docker.internal/registry" as its URI, but it should be something like "https://registry.stage.geocloud.sk/registry"

A few questions to try and replicate your problem:

- How have you deployed your instance, Docker or Tomcat?
- Have you done a new installation or just deployed the new version?

Thanks in advance

**Comment by MartinTuchyna** [comment]
Hi @oruscalleda

regarding your questions:

How have you deployed your instance, Docker or Tomcat? - Docker
Have you done a new installation or just deployed the new version? - New instalation

After some modifications, we were able to make initial registers available on the landing page:

https://registry.stage.geocloud.sk/registry

Meamwhile we have encountered another issue #259, but lets close this topic.
